README.md

```diff
@@ -49,29 +49,72 @@ polygon detection is performed and checked if the the number of sides is more th
+* So we decided to utilize transfer learning, and retrain the models on images extracted from the simulator, using 3 classes/labels only; Red, Yellow and Green.
+* We choose the "ssd_inception_v2_coco model", since it proved to be a good compromise between speed and accuracy, and retrained it on the simulator images dataset provided by Alex Lechner [here](https://github.com/alex-lechner/Traffic-Light-Classification/blob/master/README.md#1-the-lazy-approach).
+
+Sample dataset for simulator images
+![simulator training images](report/sim_visualization.png)
+
+* The configuration parameters for retraining was: 
+    * num_classes: 3.
+    * fixed_shape_resizer: 300x300, to reduce training time, since using larger image sizes during training didn't seem to increase the inference accuracy.
+    * Dropout: True.
+    * batch_size: 24.
+    * num_steps: 20000, which experimentally proved to lead to good results.
+* The training took around 3.5 hours on an NVIDIA GTX 1070 (tensorflow-gpu == 1.4.0), and the final training loss was around 2.x.
+* The retraining of the model lead to very good results; confidence levels reaching up to 0.999 even when the car is very far away from the traffic light:
+
+Here are the results of our trained model.
+(Insert image here!)
+
+#### Pros
+* This approach is very accurate.
+* It can detect all 3 colors; Red, Yellow & Green, with great confidence.
+* It can pinpoint the exact position and size of the lights, which can be further utilized for accurately calculating the stopping line position.
+
+#### Cons
+* It's not very fast, simulator FPS was limited to 10 FPS on average.
 
-We attempted transfer learning using the pre-trained SSD_inception_v2 model trained on COCO dataset, 
-and retrain it on our own dataset for NUM_EPOCHS, achieving a final loss of FINAL_LOSS.
+### Real World (Test Lot) --- SSD Approach
+* We tested the pretrained models without retraining, on real world images from the ROS bags provided by Udacity, which lead to some success, since COCO dataset already has a Traffic Light class (No.10), however it was a limited success since the ROS bags images had unusual lighting; very bright in some cases, and often the 3 light colors were not distinguishable from one another and all looked somewhat yellow.
+* Similarly we opted for retraining the "ssd_inception_v2_coco" model, but this time we compiled our own dataset, since datasets found online didn't lead to good enough results, so we labeled images from 3 different ROS bags provided by Udacity and added images from Bosch Small Traffic Lights Dataset [here](https://hci.iwr.uni-heidelberg.de/node/6132), which helped the model generalize better, and increased the detection confidence specially for instances when the traffic light was far away, since most images in the ROS bags have the traffic light in close proximity.
 
 Here is a sample of the dataset.
 ![Udacity Test Site training images](report/udacity_visualization.png)
```

Comment:
We need to replace this image with images with annotated bounding boxes in the future. Please, add TODO here to make the search places-to-update in the future.

Reply:
Isn't this a sample for the dataset images?....the annotation should be done in line: 75 & 103

Reply:
Agree, my bad. Just place a few images here in 1 or 2 lines. 12 example images is too many.